src/events/messageDelete.ts

```diff
@@ -44,7 +51,7 @@ async function messageDeleteHandler(message) {
 		},
 		{
 			name: 'Channel Name',
-			value: message.channel.name
+			value: (message.channel as any).name // TODO a better way of doing this?
```

Comment:
What is the issue here?

Reply:
Not all channels have a `name` property, and there was no obvious type predicate I could use to narrow the type to one which has a name. As the comment states I'd really like any thoughts on a better way of doing this 😂 but at least this is the same as what's there now

Reply:
I went over the types in DJS and it looks like `BaseGuildTextChannel` is what we'd want here. This event should only be sent for channels which can send messages, which would be text channels, threads, and forum channels. Text channels inherit from `BaseGuildTextChannel` directly, forum channels inherit from `ThreadOnlyChannel` and implements `BaseGuildTextChannel`, and threads inherit from `BaseChannel` and implements `BaseGuildTextChannel`

`GuildChannel` also looks promising as it represents many different channel types already, but I didn't see mentions of threads anywhere

That's assuming that these types are exported though. If not then either using `any` still or just making our own type is our only option

Reply:
Something like this?
```ts
	let channelName = 'No channel name found';
	if (message.channel.type === ChannelType.GuildText) {
		channelName = message.channel.name;
	}
```

Reply:
I was specifically thinking about an `instanceof` check, but if that works then that's fine too. Though it can be condensed with a ternary

Reply:
Actually no, that alone doesn't seem like it would work. I may have to go back and check the docs though. DJS separates classes like "text channel" and "thread" and "forum channel", so *only* checking for a text channel may not be enough here

That's why I was thinking of an `instanceof` check, as that would catch *any* channel type which uses `BaseGuildTextChannel` in it's hierarchy

Reply:
Seems like `BaseGuildTextChannel` doesn't exist in v14. I think that was from v12, it only exists in the docs in the upgrading guide from 12->13?

Reply:
I've updated this with the `instanceof` check and I've tested it with deleted messages in a regular channel, forum channel, and thread and they all worked fine